### PR TITLE
Change /usr/portage council approved locations (bug 378603)

### PR DESCRIPTION
**Pull request by zmedico**
This includes a _compat_upgrade.default_locations script that the
ebuild can call in pkg_preinst in order to maintain backward-compatible
defaults when appropriate. The new defaults are specified in the
summary of the 20180729 council meeting:

    Vote: Default locations for the Gentoo repository, distfiles, and
    binary packages will be, respectively:
       /var/db/repos/gentoo
       /var/cache/distfiles
       /var/cache/binpkgs
    Accepted with 6 yes votes and 1 no vote.

See: https://projects.gentoo.org/council/meeting-logs/20180729-summary.txt
Bug: https://bugs.gentoo.org/378603

### CODE REVIEW

File: cnf/make.conf.example [hunk 1]
```diff
@@ -111,24 +111,24 @@
...
 #     emerges. After packages are built, it is safe to remove any and
 #     all files from this directory since they will be automatically
 #     fetched on demand for a given build. If you would like to
 #     selectively prune obsolete files from this directory, see
 #     eclean from the gentoolkit package. Note that locations under
-#     /usr/portage are not necessarily safe for data storage. See the
+#     /var/db/repos/gentoo are not necessarily safe for data storage. See the
 #     PORTDIR documentation for more information.
-#DISTDIR=/usr/portage/distfiles
+#DISTDIR=/var/db/repos/gentoo/distfiles
```
Comment (nE0sIghT) [inline cnf/make.conf.example]: Should be ```/var/cache/distfiles```

File: cnf/make.conf.example [hunk 2]
```diff
@@ -111,24 +111,24 @@
...
 #
 # PKGDIR is the location of binary packages that you can have created
 #     with '--buildpkg' or '-b' while emerging a package. This can get
 #     up to several hundred megs, or even a few gigs. Note that
-#     locations under /usr/portage are not necessarily safe for data
+#     locations under /var/db/repos/gentoo are not necessarily safe for data
 #     storage. See the PORTDIR documentation for more information.
-#PKGDIR=/usr/portage/packages
+#PKGDIR=/var/db/repos/gentoo/packages
```
Comment (nE0sIghT) [inline cnf/make.conf.example]: Should be ```/var/cache/binpkgs```

File: lib/portage/cache/flat_hash.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def __iter__(self):
 					# Only recurse 1 deep, in order to avoid iteration over
 					# entries from another nested cache instance. This can
 					# happen if the user nests an overlay inside
-					# /usr/portage/local as in bug #302764.
+					# /var/db/repos/gentoo/local as in bug #302764.
```
Comment (nE0sIghT) [inline lib/portage/cache/flat_hash.py]: IMHO nested local repos should be unsupported now

**Comment by bl33pbl0p** [comment]
Anything keeping this from getting merged?

**Comment by ulm** [comment]
Shouldn't emerge-webrsync be updated too?
It seems a bit strange if the snapshot of the gentoo repository is still named portage-${date}.tar.${compression} and unpacks into "portage" as top level dir.

File: cnf/make.conf.example [hunk 4]
```diff
@@ -111,24 +111,24 @@
...
 #     fetched on demand for a given build. If you would like to
 #     selectively prune obsolete files from this directory, see
 #     eclean from the gentoolkit package. Note that locations under
-#     /usr/portage are not necessarily safe for data storage. See the
-#     PORTDIR documentation for more information.
-#DISTDIR=/usr/portage/distfiles
+#     /var/db/repos/gentoo are not necessarily safe for data storage.
+#     See the PORTDIR documentation for more information.
+#DISTDIR=/var/cache/distfiles
 #
```
Comment (ulm) [inline cnf/make.conf.example]: I believe it is unlikely that anyone would change the DISTDIR location to a subdir of /var/db/repos/gentoo. So the following two sentences are more confusing than helpful, and IMHO should be deleted entirely:
"Note that locations under /var/db/repos/gentoo are not necessarily safe for data storage. See the PORTDIR documentation for more information."

File: cnf/make.conf.example [hunk 5]
```diff
@@ -111,24 +111,24 @@
...
 # PKGDIR is the location of binary packages that you can have created
 #     with '--buildpkg' or '-b' while emerging a package. This can get
 #     up to several hundred megs, or even a few gigs. Note that
-#     locations under /usr/portage are not necessarily safe for data
-#     storage. See the PORTDIR documentation for more information.
-#PKGDIR=/usr/portage/packages
+#     locations under /var/db/repos/gentoo are not necessarily safe for
+#     data storage. See the PORTDIR documentation for more information.
+#PKGDIR=/var/cache/binpkgs
 #
```
Comment (ulm) [inline cnf/make.conf.example]: Ditto.

File: man/make.conf.5 [hunk 6]
```diff
@@ -219,10 +219,10 @@ Use the \fBPORTAGE_RO_DISTDIRS\fR variable to specify one or
 more read-only directories containing distfiles.
 
 Note
-that locations under /usr/portage are not necessarily safe for data storage.
+that locations under /var/db/repos/gentoo are not necessarily safe for data storage.
 See the \fBPORTDIR\fR documentation for more information.
```
Comment (ulm) [inline man/make.conf.5]: Ditto.

File: man/make.conf.5 [hunk 7]
```diff
@@ -800,10 +800,10 @@ to it's category. However, for backward compatibility with the layout
 used by older versions of portage, if the \fI${PKGDIR}/All\fR directory
 exists then all packages will be stored inside of it and symlinks to
 the packages will be created in the category subdirectories. Note
-that locations under /usr/portage are not necessarily safe for data storage.
+that locations under /var/db/repos/gentoo are not necessarily safe for data storage.
 See the \fBPORTDIR\fR documentation for more information.
```
Comment (ulm) [inline man/make.conf.5]: Ditto.

**Comment by mgorny** [comment]
Ping.

File: cnf/make.conf.example (hunk 4 above)
Comment (zmedico) [reply cnf/make.conf.example]: done

File: cnf/make.conf.example (hunk 5 above)
Comment (zmedico) [reply cnf/make.conf.example]: done

**Comment by zmedico** [comment]
I'd like to merge this an release it today.

**Comment by zmedico** [comment]
Shouldn't emerge-webrsync be updated too?
    It seems a bit strange if the snapshot of the gentoo repository is still named portage-${date}.tar.${compression} and unpacks into "portage" as top level dir.

That's out of scope, since the code already works for arbitrary repository locations.

**Pull request merged**